frontend/rust-lib/flowy-document2/src/manager.rs

```diff
@@ -32,23 +33,45 @@ impl DocumentManager {
     }
   }
 
+  pub fn create_document(&self, doc_id: String, data: DocumentDataWrapper) -> FlowyResult<Arc<Document>> {
+    self.get_document(doc_id, Some(data))
+  }
+
+  fn get_document(&self, doc_id: String, data: Option<DocumentDataWrapper>) -> FlowyResult<Arc<Document>> {
+    let collab = self.get_collab_for_doc_id(&doc_id)?;
+    let document = Arc::new(Document::new(collab)?);
+    self.documents.write().insert(doc_id, document.clone());
```

Comment:
I think we should put this line only in the `open_document` function, because we want to keep the document alive only when we need to listen to its callback.

Reply:
use `if let` will be better.

if let Some(data) = data { ... }